### ISSUE TITLE
Missing distr_param argument in randu

### ISSUE THREAD

**Issue by MOOnLyer** [RcppCore/RcppArmadillo#382]
I just noticed that we cannot specify distr_param for randu. Then I compare the armadillo_bits in RcppArmadillo to the one from [http://arma.sourceforge.net/](url). It seems that RcppArmadillo's version removes functions like 
```
double randu(const distr_param& param) {}
```
I wonder why we choose not to include those function.

Thanks!

**Comment by eddelbuettel** [comment]
Please check the release history and notes; RcppArmadillo is set up to prefer the R RNG as we can then share the seeding.  So what you noted is both correct -- and documented and intentional.   So I think we can close this.

**Comment by MOOnLyer** [comment]
@eddelbuettel Thanks for the reply.

> Please check the release history and notes; 

I tried to go through the Changlog to see if anything relevant, and found nothing. Would you kindly point it out? 

> RcppArmadillo is set up to prefer the R RNG as we can then share the seeding.

I am a little confused. R's runif is able to take in (min and max) as its argument, isn't it?

**Comment by eddelbuettel** [comment]
Search in the [NEWS.Rd file](https://github.com/RcppCore/RcppArmadillo/blob/master/inst/NEWS.Rd) 

You confuse a _seed_ with _parameters_.  What we are after is RNG behaviour:

```r
> set.seed(42L)  # seeding
> runif(3)   # calling R RNG
[1] 0.914806 0.937075 0.286140
> 
> Rcpp::cppFunction("arma::vec my_runif(int n) { return arma::randu(n); }", depends="RcppArmadillo")
> set.seed(42L)    # seeding again
> my_runif(3)      # calling Arma ... which is mapped to R's RNG
         [,1]
[1,] 0.914806
[2,] 0.937075
[3,] 0.286140
> 
```

That we get the same draws is a feature.  You are able to turn this off with a define if you prefer but what I show is the _intended_ default behaviour.

**Comment by eddelbuettel** [comment]
May not be documented in NEWS.Rd after all. It's been like this forever, though, and I do not have time to search the docs for you now.  Sorry.

**Comment by conradsnicta** [comment]
@MOOnLyer The distr_param() option has been added to randu() and randn() in Armadillo 11.2, which was released a week ago.  This has been propagated to RcppArmadillo 0.11.2.x, which is available on CRAN as of yesterday.

The changelogs are at:
* http://arma.sourceforge.net/docs.html#changelog
* http://dirk.eddelbuettel.com/blog/2022/06/15/#rcpparmadillo_0.11.2.0.0

The relevant change in both logs is: "extended [randu()](http://arma.sourceforge.net/docs.html#randu) and [randn()](http://arma.sourceforge.net/docs.html#randn) to allow specification of distribution parameters"

**Comment by eddelbuettel** [comment]
Conrad, that is both true and irrelevant as I said earlier as _RcppArmadillo by default switches to the 'alternate' RNGs offered by R_.  My example _still_ stands:

```
> packageVersion("RcppArmadillo")
[1] ‘0.11.2.0.0’
> Rcpp::cppFunction("arma::vec my_runif(int n) { return arma::randu(n); }", depends="RcppArmadillo")
> set.seed(42L)    # seeding again
> my_runif(3)      # calling Arma ... which is mapped to R's RNG
         [,1]
[1,] 0.914806
[2,] 0.937075
[3,] 0.286140
> 
```

**Comment by conradsnicta** [comment]
That's fine. The alternate RNG is still used in all forms of randu() and randn(), in all versions of Armadillo that are used in their RcppArmadillo guise.

The issue by @MOOnLyer was the apparent lack of randu() and randn() forms that accept distr_param() arguments, which is now resolved -- the new forms are in RcppArmadillo 0.11.2.x.  This issue is not related to the internal use of the alternate RNG.
